tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15277,9 +15371,16 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr) {
 
 // if this is the Entry FD, then try adding the target profile
 // shader attribute to the FD and carry on with validation
-void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD) {
+void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD, bool &isActiveEntry) {
+  isActiveEntry = false;
   const std::string &EntryPointName = S.getLangOpts().HLSLEntryFunction;
 
+  if (S.getLangOpts().IsHLSLLibrary) {
+    // TODO: Improve accuracy by analyzing -exports option
+    // to determine which entries are active for lib target
+    isActiveEntry = true;
```

Comment:
I think this part needs to be defaulted and commented about in `DiagnoseEntry`.

Reply:
In fact, if we have a library target, we wouldn't need to call this function at all.